pydantic/main.py

```diff
@@ -228,15 +227,17 @@ class BaseModel(metaclass=MetaModel):
     Config = BaseConfig
     __slots__ = ('__values__', '__fields_set__')
 
-    def __init__(self, **data: Any) -> None:
+    def __init__(self, _object: Any = None, **data: Any) -> None:
```

Comment:
I'm trying to maintain a hard rule that only data is passed to `__init__` to end users using pydantic APIs being able to achieve unexpected behaviour by setting special fields.

In this case every use of pydantic models what didn't want to use `_object ` would need a manual check before calling `Model(**mydata)` that `mydata` doesn't include `_object`.

Better to add another `classmethod` a bit like `construct` that creates a new model instead without calling `__init__` (or maybe does use `__init__`?), but leave `__init__`'s signature unchanged.

Reply:
Here the problem is that `Field`'s implementation instantiates the class directly (calling `__init__`), then we would need checks in `Field`s validators to check which method to call in several places.

But I just realized that this could be achieved with `*args`, that way we avoid that problem. Let me try implementing that and see if that works.

Reply:
You mean `MyModel` accepting `(*args, **kwargs)` like dataclasses?

I'm not opposed to that but I think it might end up being a big change. Maybe worth doing it as a separate PR before this?

More generally if you you can use `*args`, maybe you'd be best to just use pydantic dataclasses for all of this, then `Modal` could maybe remain (mostly) unchanged?

Reply:
I just updated it. Nope, no special support. It only avoids the problem of `Model(_object="foo", b="bar")`.

`*args` is used only to avoid having a possible keyword argument. The first value would be extracted from `args`, only when there's no `**data`. But it doesn't add any special support for `*args`. And actually, only the first value is used. But by having `*args`, it can't be passed as a keyword argument.

So, it would keep working as normally, with `Model(a="foo", b="bar")`.

But when using `Model(my_object)`, `my_object` would be taken from the `*args`, only the first element.

So, `Model(_object="foo", b="bar")` wouldn't be a problem.

---

I can't use only dataclasses because what I need is to support it when `Field.validate` creates a model from a class.

 `Field.validate` calls `Field._validate_singleton`, etc.

They then call `Field.apply_validators`.

And then those will call (later at some point) `cls(data)`. With `cls` being a Pydantic model (normally a standard model inheriting from `BaseModel`).

But by implementing it this way, it has the effect that it can be used in many other cases (with ORMs or other data objects, standard dataclasses, etc).